Program.cs

```diff
@@ -1,38 +1,17 @@
-﻿using Laserfiche.Api.Client.OAuth;
-using Newtonsoft.Json;
-using System.Diagnostics;
-using System.Text;
-
-namespace Laserfiche.Repository.Api.Client.Sample.ServiceApp
+﻿namespace Laserfiche.Repository.Api.Client.Sample.ServiceApp
 {
     static class Program
     {
         public static async Task Main()
         {
-            // Read credentials from file system
-            var readConfigFileOk = Utils.LoadFromDotEnv("TestConfig.env");
-            if (!readConfigFileOk)
-            {
-                Trace.TraceWarning("Failed to read credentials.");
-                return;
-            }
-
-            // Read credentials from envrionment
-            var servicePrincipalKey = Environment.GetEnvironmentVariable("SERVICE_PRINCIPAL_KEY");
-            var base64EncodedAccessKey = Environment.GetEnvironmentVariable("ACCESS_KEY");
-            if (base64EncodedAccessKey == null)
-            {
-                throw new InvalidOperationException("Cannot continue due to missing access key.");
-            }
-            var decoded = Encoding.UTF8.GetString(Convert.FromBase64String(base64EncodedAccessKey));
-            var accessKey = JsonConvert.DeserializeObject<AccessKey>(decoded);
-            var repositoryId = Environment.GetEnvironmentVariable("REPOSITORY_ID");
+            // Get credentials
+            var config = new ServiceConfig("TestConfig.env");
 
             // Create the client
-            var repoClient = RepositoryApiClient.Create(servicePrincipalKey, accessKey);
+            var repoClient = RepositoryApiClient.Create(config.ServicePrincipalKey, config.AccessKey);
 
             var rootEntryId = 1;
-            var entryListing = await repoClient.EntriesClient.GetEntryListingAsync(repositoryId, rootEntryId);
+            var entryListing = await repoClient.EntriesClient.GetEntryListingAsync(config.RepositoryId, rootEntryId);
```

Comment:
the program in the js sample app also prints out the repo name, root folder along with its root folder children, should it also be done in the C# counter part sample project?

Reply:
Sorry I only committed the change but forgot to push it. It's updated now.